src/error.rs

```diff
@@ -1,7 +1,52 @@
 //! Contains error types and other error handling tools.
 
+use core::fmt;
+
+use crate::consensus::encode;
+use crate::internal_macros::write_err;
 pub use crate::parse::ParseIntError;
 
+/// A general error code, other errors should implement conversions to/from this
```

Comment:
It's not an error *code*, it's an error type.

However the doc contradicts itself here. It hides from impl which mean "do not use" and here it documents there are conversions. (IMO we should either commit to conversions and unhide them or remove them because they are too easy to use by accident.)

If you think it's better to blindly move things without changing for this kind of PR, that's perfectly OK for me.

Reply:
I've wondered this a bit lately, right now I think the best approach is move things as they are (with formatting changes) to flatten util, then we can start crate smashing - do that using move only as well. And then when things are in separate crates improve each crate pushing each towards 1.0 (i.e., stabilize API).